### PR TITLE
Add editor styles to product editor and iframe editor

### LINKED ISSUE #38021: Iframe Editor: Missing button styles in editor
Originally noted in [this comment](https://github.com/woocommerce/woocommerce/pull/37990#pullrequestreview-1401352669) the button styles are missing inside the iframe editor component.

### Screenshots

Image

### Reproducing

1. Enable the new product blocks experience
2. Navigate to Products -> Add new
3. Click "Add description"
4. Add a button block
5. Note that styles are missing

### Acceptance criteria
* [ ] The button styles should appear when editing content in the modal editor
* [ ] The button styles should appear in the preview area after saving a description

### PR DESCRIPTION
**Pull request by joshuatf**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Fixes an issue where certain editor styles (such as buttons) were not being applied to the editor and iframe editor.

#### Before
Screen Shot 2023-05-08 at 9 50 14 AM

#### After

Screen Shot 2023-05-08 at 9 49 08 AM

Closes #38021.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Navigate to Tools -> WCA Test Helper -> Features and enable the new product blocks editing experience
2. Navigate to Products -> Add new
3. Click on the "Add description" button
4. Add a button block
5. Note that the button styles appear using your site default styles in the modal/iframe editor
6. Close the modal
7. Note that the preview also shows the same styles

### CODE REVIEW

File: packages/js/product-editor/src/components/block-editor/block-editor.tsx [hunk 1]
```diff
@@ -101,6 +104,8 @@ export function BlockEditor( {
 					onChange={ onChange }
 					settings={ settings }
 				>
+					<EditorStyles styles={ settings?.styles } />
+					{/* <EditorStyles styles={ styles } /> */}
```
Comment (louwie17) [inline packages/js/product-editor/src/components/block-editor/block-editor.tsx]: This line can be removed right?

**Review by louwie17 — COMMENTED** [review]
Nice work, this is testing well!
There are a couple lint errors and I left one inline comment.

Also I noticed one issue we will have to create a follow up for, the `content-preview` doesn't render any of the inline css, so if I set the layout to `center` it should add in-line css of `display: flex; justifyContent: center`, but it only adds that as html comments:
`<!-- wp:buttons {"layout":{"type":"flex","justifyContent":"center"}} -->`

Maybe there is an alternative serialize function for this?

File: packages/js/product-editor/src/components/block-editor/block-editor.tsx (hunk 1 above)
Comment (joshuatf) [reply packages/js/product-editor/src/components/block-editor/block-editor.tsx]: It can!  Good catch

**Comment by joshuatf** [comment]
>Also I noticed one issue we will have to create a follow up for, the content-preview doesn't render any of the inline css, so if I set the layout to center it should add in-line css of display: flex; justifyContent: center, but it only adds that as html comments:

Good point.  Will create another issue to address this.

**Review by louwie17 — APPROVED** [review]
LGTM 🚀

**Pull request merged**